File: DSS/DSS.Common/DSS.Common.ViewModels.Common/Documents/Display/DisplayDocumentViewModel.cs
using System;
using System.Collections.Generic;

namespace DSS.Common.ViewModels.Documents
{
    /// <summary>
    /// Class used as a simple display for the documents in the system.
    /// </summary>
    public class DisplayDocumentViewModel
    {
        /// <summary>
        /// The document id
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// The title of the document
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// The Document Description entered when the document was uploaded.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// The name of the author on the initial document
        /// </summary>
        public string AuthorName { get; set; }

        /// <summary>
        /// Indicator wether the document is indexed in lucene
        /// </summary>
        public bool IsIndexed { get; set; }

        /// <summary>
        /// The list of categories for the document
        /// </summary>
        public List<string> Categories { get; set; }

        /// <summary>
        /// The list of keywords for the document
        /// </summary>
        public List<string> Keywords { get; set; }
    }
}
 

File: DSS/DSS.Presentation/DSS.Presentation.Web/Scripts/src/Ko/ViewModels/Document/DisplayDocumentViewModel.js
// The DisplayDocument client side knockout view model used to display documents.
(function() {
    var displayDocumentViewModel = function(id, title, authorName, isIndexed, keywords, categories) {
        // the id of the document
        this.id = ko.observable(id);

        // the actual document title
        this.title = ko.observable(title);

        // the name of the document title
        this.authorName = ko.observable(authorName);
        
        // indicates if the document is indexed
        this.isIndexed = ko.observable(isIndexed);

        // the document keywords
        this.keywords = ko.observableArray(keywords);

        // the document categories
        this.categories = ko.observableArray(categories);
    };

    // Namespace the document display view model
    DSS.namespace("Ko.ViewModels.Document.DisplayDocumentViewModel", displayDocumentViewModel);
})();